### PR TITLE
Add needed structs for fd passing

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

**Comment by alexcrichton** [comment]
Looks good to me, but seems CI is failing?

**Comment by GuillaumeGomez** [comment]
Seems like fields are missing, strange. Taking a deeper look.

**Comment by GuillaumeGomez** [comment]
All fixed!

**Comment by GuillaumeGomez** [comment]
But maybe not (I added missing const).

**Comment by GuillaumeGomez** [comment]
Ok, I forget about consts, too much troubles.

File: src/unix/notbsd/linux/musl/b64/mod.rs [hunk 1]
```diff
@@ -114,6 +114,13 @@ s! {
         pub msg_flags: ::c_int,
     }
 
+    pub struct cmsghdr {
+        pub cmsg_len: ::socklen_t,
+        pub __pad1: ::c_int,
```
Comment (semarie) [inline src/unix/notbsd/linux/musl/b64/mod.rs]: does `__pad1` need to be public ?

Comment (GuillaumeGomez) [inline src/unix/notbsd/linux/musl/b64/mod.rs]: I think it should. After all it's a C binding, so people can do whatever they want.

**Pull request merged**